frontend/src/components/workspace/EditSavedSearchModal.tsx

```diff
@@ -122,6 +129,7 @@ export function EditSavedSearchModal({
   const router = useRouter();
 
   const [validationError, setValidationError] = useState<string>();
+  const [originalName] = useState<string>(queryName);
```

Comment:
this is a good idea, but if you rename a query more than once, the first name of the query will continue to show up, rather than updating to subsequent renames. Using `usePrevious` handles that edge case
```suggestion
  const originalName = usePrevious(queryName);
```